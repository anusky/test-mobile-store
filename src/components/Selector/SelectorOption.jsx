import PropTypes from "prop-types";

const SelectorOption = ({ label, value }) => {
  return <option value={value}>{label}</option>;
};

export default SelectorOption;
SelectorOption.propTypes = {
  label: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};
