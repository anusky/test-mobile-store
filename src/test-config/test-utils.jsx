import "./__mocks__/intersectionObserverMock";

import { render } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import PropTypes from "prop-types";

import productInfo from "./__fixtures__/product.json";
import productList from "./__fixtures__/productList.json";
import cart from "./__fixtures__/cart.json";
import { ProductContextProvider } from "src/context/productContext";
import { PAGE_STRUCTURE } from "src/utils/constants";

const customRender = (ui, options = {}) => {
  const { routerInitialIndex = 0 } = options;

  const AllTheProviders = ({ children }) => {
    const routes = [
      {
        path: "/",
        element: children,
        id: PAGE_STRUCTURE.home.clientPageId,
        loader: () => ({
          productList,
          cart,
        }),
        handle: {
          crumb: () => ({ url: "/", content: "home route" }),
        },
        children: [
          {
            path: "/:productId",
            element: children,
            id: PAGE_STRUCTURE.product.clientPageId,
            loader: () => productInfo,
            handle: {
              crumb: () => ({ url: "/", content: "product route" }),
            },
          },
        ],
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ["/", "/123"],
      initialIndex: routerInitialIndex,
    });
    return (
      <RouterProvider router={router}>
        <ProductContextProvider
          value={{
            list: productList,
            cart: cart,
          }}
        >
          {children}
        </ProductContextProvider>
      </RouterProvider>
    );
  };

  AllTheProviders.propTypes = {
    children: PropTypes.node.isRequired,
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
export { customRender as render };
