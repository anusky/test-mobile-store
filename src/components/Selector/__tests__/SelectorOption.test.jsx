import { act } from "react-dom/test-utils";
import { render, screen } from "src/test-config/test-utils";

import SelectorOption from "../SelectorOption";

describe("SelectorOption coverage", () => {
  const props = { label: "test name", value: 123 };

  it("Should render correctly", async () => {
    render(<SelectorOption {...props} />);

    expect(await screen.findByRole("option")).toBeInTheDocument();
  });

  it("matches snapshot ", async () => {
    let container;
    await act(() => {
      container = render(<SelectorOption {...props} />);
    });
    expect(container.asFragment()).toMatchSnapshot();
  });
});
