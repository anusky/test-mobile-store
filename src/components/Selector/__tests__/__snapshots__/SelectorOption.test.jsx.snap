// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectorOption coverage matches snapshot  1`] = `
<DocumentFragment>
  <option
    value="123"
  >
    test name
  </option>
</DocumentFragment>
`;
